addons/account/tests/test_account_payment.py

```diff
@@ -852,3 +852,51 @@ def test_suggested_default_partner_bank(self):
             'available_partner_bank_ids': self.comp_bank_account2.ids,
             'partner_bank_id': self.comp_bank_account2.id,
         }])
+
+    def test_internal_bank_transfer(self):
+        """The purpose of this test is to check that the right accounts are computed when making an internal bank transfer"""
+
+        # We first check the behavior when no receipts/payments accounts are set on bank journals
+        company = self.company_data['default_journal_bank'].company_id
+
+        bank = self.bank_journal_1
+        bank_2 = self.bank_journal_2
+
+        with Form(self.env['account.payment']) as payment_form:
+            payment_form.is_internal_transfer = True
+            payment_form.payment_type = 'outbound'
+            payment_form.amount = 100.0
+            payment_form.journal_id = bank
+            payment_form.destination_journal_id = bank_2
+        payment = payment_form.save()
```

Comment:
you can't do that because you are saving twice the same payment (one time when exiting the `with`, one time when calling `save`):
```py
payment_form = Form(self.env['account.payment'])
payment_form.is_internal_transfer = True
payment_form.payment_type = 'outbound'
payment_form.amount = 100.0
payment_form.journal_id = bank
payment_form.destination_journal_id = bank_2
payment = payment_form.save()
```
But you can create it directly with a `create`, right? (this is far more efficient)